src/Stratis.Bitcoin.Features.Api/Program.cs

```diff
@@ -9,18 +11,26 @@ namespace Stratis.Bitcoin.Features.Api
 {
     public class Program
     {
-        public static void Main(string[] args)
-        {
-        }
-
-        public static IWebHost Initialize(IEnumerable<ServiceDescriptor> services, FullNode fullNode, ApiSettings apiSettings)
+        public static IWebHost Initialize(IEnumerable<ServiceDescriptor> services, FullNode fullNode,
+            ApiSettings apiSettings, ICertificateStore store, IWebHostBuilder webHostBuilder = null)
         {
             Guard.NotNull(fullNode, nameof(fullNode));
 
             Uri apiUri = apiSettings.ApiUri;
 
-            IWebHost host = new WebHostBuilder()
-                .UseKestrel()
+            X509Certificate2 certificate = apiSettings.UseHttps ? GetHttpsCertificate(apiSettings, store) : null;
+
+            webHostBuilder = webHostBuilder ?? new WebHostBuilder();
+
+            webHostBuilder
+                .UseKestrel(options =>
+                    {
+                        if (!apiSettings.UseHttps) return;
+                        options.Listen(
+                            IPAddress.Any,
```

Comment:
This is better, but it still is not respecting bind addresses

Reply:
@Aprogiena Thanks for feedback, wondering if you think this is better ?

```
            webHostBuilder
                .UseKestrel(options =>
                    {
                        if (!apiSettings.UseHttps) return;
                        Action<ListenOptions> configureListener = listenOptions => { listenOptions.UseHttps(certificate); };
                        var ipAddresses = Dns.GetHostAddresses(apiSettings.ApiUri.DnsSafeHost);
                        foreach (var ipAddress in ipAddresses)
                        {
                            options.Listen(ipAddress, apiSettings.ApiPort, listenOptions => listenOptions.UseHttps(certificate));
                        }
                    })
```

Or are you thinking of getting the "bind" addresses from the ConnectionManagerSettings, probably something like that
```
this.Listen.AddRange(config.GetAll("bind")
        .Select(c => new NodeServerEndpoint(c.ToIPEndPoint(port), false)));
```

 and then do the listen on them ?

Reply:
I'm not familiar with Kestrel, so I cannot advise on the exact implementation, but i meant `this.Listen` On connection manager settings. i don't understand `var ipAddresses = Dns.GetHostAddresses(apiSettings.ApiUri.DnsSafeHost);`  - Why would we need that here